autoinstrumentation/nodejs/src/autoinstrumentation.ts

```diff
@@ -12,6 +13,19 @@ import { diag } from '@opentelemetry/api';
 
 import { NodeSDK } from '@opentelemetry/sdk-node';
 
+function getTraceExporter() {
+    switch (process.env.OTEL_EXPORTER_OTLP_PROTOCOL) {
+        case undefined:
+        case '':
+        case 'grpc':
+            return new OTLPTraceExporter();
+        case 'http/protobuf':
+            return new OTLPTraceExporterHTTP();
```

Comment:
This is incorrect. The OTLP exporter from `@opentelemetry/exporter-trace-otlp-http` is actually the exporter for `http/json`, and not for `http/proto`. Granted the "http" in the package name is certainly misleading. The exporter package for `http/proto` is `@opentelemetry/exporter-trace-otlp-proto`.

Seeing the equivalent handling code in `@opentelemetry/sdk-node` is helpful:
https://github.com/open-telemetry/opentelemetry-js/blob/eb3ca4fb07ee31c62093f5fcec56575573c902ce/experimental/packages/opentelemetry-sdk-node/src/TracerProviderWithEnvExporter.ts#L31-L57

Options:
1. Copy the equivalent of that code from the `sdk-node` package.
2. Or, another option (which may need more discussion, separate from this PR) might be to *remove* specifying `traceExporter` to the `NodeSDK` constructor, and let the NodeSDK handle selecting the right one. This however, then brings up the question of what the default exporter should be, which impacts the Operator configuration.

## changing the default exporter to `http/proto`?

By default the `NodeSDK` will use the `http/proto` trace exporter. Over time `http/proto` has become the default OTLP protocol used by most OTel languages. You can see this somewhat from the YAML configuration for the OTel Operator `Instrumentation` given here: https://github.com/open-telemetry/opentelemetry-operator/blob/main/README.md#opentelemetry-auto-instrumentation-injection

It sets the default exporter to `http://otel-collector:4317`, which is gRPC, but then has custom env for most languages to change to `http://otel-collector:4318`, which is an HTTP endpoint (for receiving either `http/proto` or `http/json`).

I haven't been involved much with the OTel Operator, but I wonder if it would consider switching the default for `nodejs` over to `http://otel-collector:4318` as well. I realize that would be a breaking change.

The benefit is that "autoinstrumentation.ts" could just use the `NodeSDK` defaults, and not have to handle selecting a `traceExporter` at all. The NodeSDK's envvar handling will do the right thing.

Reply:
>  The OTLP exporter from @opentelemetry/exporter-trace-otlp-http is actually the exporter for http/json, and not for http/proto. Granted the "http" in the package name is certainly misleading

@trentm, thanks for pointing this out and giving good explanation! It turned out that Grafana Tempo that i used, works with both http/json and http/proto when setting 
> OTEL_EXPORTER_OTLP_ENDPOINT=https://tempo.myorg.com

so i changed the code to support both (I also tested both manually with Tempo).

Reply:
> I haven't been involved much with the OTel Operator, but I wonder if it would consider switching the default for nodejs over to http://otel-collector:4318 as well. I realize that would be a breaking change.
>
> The benefit is that "autoinstrumentation.ts" could just use the NodeSDK defaults, and not have to handle selecting a traceExporter at all. The NodeSDK's envvar handling will do the right thing.

That's doesn't sound like a bad idea to me in long run, but with this PR I'd keep the backwards compatibility, meaning keeping grpc as the default traces exporter as it was.

If maintainers of this project want to do the breaking change, then perhaps with separate PR, so that the breaking change doesn't block releasing this feature.

Reply:
@atsu85 would you mind opening an issue for changing the default exporter for node?

Reply:
> would you mind opening an issue for changing the default exporter for node?

I could, but maybe @trentm you want to do it yourself. Otherwise I would just copy-paste the chapter you wrote
> # [changing the default exporter to http/proto?](https://github.com/open-telemetry/opentelemetry-operator/pull/3413#discussion_r1826094997)
> ...

Reply:
I created https://github.com/open-telemetry/opentelemetry-operator/issues/3423 to discuss this separately. Thanks.